.github/workflows/main.yml

```diff
@@ -0,0 +1,61 @@
+name: GitHub Actions
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  test:
+    name: ${{ matrix.task }} (${{ matrix.os }}, ${{ matrix.compiler }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            task: r-package
+            compiler: gcc
+          - os: ubuntu-latest
+            task: r-package
+            compiler: clang
+          - os: macOS-latest
+            task: r-package
+            compiler: gcc
+          - os: macOS-latest
+            task: r-package
+            compiler: clang
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 5
+          submodules: true
+      - name: Setup and run tests on Linux and macOS
+        if: matrix.os != 'windows-latest'
+        shell: bash
+        env:
+          COMPILER: "${{ matrix.compiler }}"
+          CONDA_ENV: "test-env"
+          GITHUB_ACTIONS: "true"
+          TASK: "${{ matrix.task }}"
+        run: |
+          if [[ "${{ matrix.os }}" == "macOS-latest" ]]; then
+              export OS_NAME="macos"
+              export R_MAC_VERSION=3.6.3
+          elif [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+              export OS_NAME="linux"
+              export R_LINUX_VERSION=3.6.3-1bionic;
+          fi
+          if [ -z ${{ matrix.compiler }} ]; then
+              export COMPILER=${{ matrix.compiler }}
+          fi
```

Comment:
Sorry, can you please explain why this `if` statement is needed?

Reply:
ah! Another thing left over from when I was including the `lint` task here. That task doesn't have a `compiler`. But now that that's removed we can remove this!

https://github.com/microsoft/LightGBM/pull/3119/commits/3e378bffbd908c2a75f6c95444343a9f819e9be8